pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -168,7 +168,7 @@ impl Layer for DeltaLayer {
         lsn: Lsn,
         reconstruct_data: &mut PageReconstructData,
     ) -> Result<PageReconstructResult> {
-        let mut cont_lsn: Option<Lsn> = Some(lsn);
+        let mut need_image = true;
```

Comment:
Is `need_image` always equal to `reconstruct_data.page_img.is_none()`

Reply:
No, you will also have `need_image==false` if the oldest record returned in 'reconstruct_data.records' has `will_init==true`.

Reply:
Ahh right. Thanks